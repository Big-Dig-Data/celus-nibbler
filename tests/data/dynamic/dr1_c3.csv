Database Report 1(R3) ,Total Searches and Sessions by Month and Database 
MY -- Institution
Date run:
2023-01-01
,Publisher,Platform,, Jan-2021, Feb-2021, Mar-2021, Apr-2021, May-2021, Jun-2021, Jul-2021, Aug-2021, Sep-2021, Oct-2021, Nov-2021, Dec-2021,YTD Total
DB1,Publisher1,Platform1,Total searches run,0,0,3,0,0,0,0,0,1,0,1,0,5
DB1,Publisher1,Platform1,Searches-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
DB1,Publisher1,Platform1,Total sessions,1,3,2,0,0,0,0,0,2,0,1,1,10
DB1,Publisher1,Platform1,Sessions-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
,,,,,,,,,,,,
DB2, ,Platform1,Total searches run,0,0,2,0,0,0,0,0,1,0,0,0,3
DB2, ,Platform1,Searches-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
DB2, ,Platform1,Total sessions,1,3,3,0,0,0,0,0,1,0,1,0,9
DB2, ,Platform1,Sessions-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
,,,,,,,,,,,,
DB3,Publisher1,Platform1,Total searches run,0,0,0,0,0,0,0,0,0,0,0,0,0
DB3,Publisher1,Platform1,Searches-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
DB3,Publisher1,Platform1,Total sessions,1,3,2,0,0,0,0,0,1,1,2,0,10
DB3,Publisher1,Platform1,Sessions-federated and automated,0,0,0,0,0,0,0,0,0,0,0,0,0
,,,,,,,,,,,,
