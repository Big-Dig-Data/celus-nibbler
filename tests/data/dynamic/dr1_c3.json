{
  "kind": "non_counter.generic",
  "version": 1,
  "parser_name": "dr1_c3",
  "data_format": {
    "name": "dr1_c3"
  },
  "platforms": [
    "Platform1"
  ],
  "metrics_to_skip": [],
  "titles_to_skip": ["EMPTY"],
  "dimensions_to_skip": {},
  "metric_aliases": [],
  "dimension_aliases": [],
  "heuristics": {
    "conds": [
      {
        "pattern": "Database Report 1",
        "coord": {
          "row": 0,
          "col": 0
        },
        "kind": "regex"
      },
      {
        "pattern": "Total Searches and Sessions by Month and Database",
        "coord": {
          "row": 0,
          "col": 1
        },
        "kind": "regex"
      }
    ],
    "kind": "and"
  },
  "areas": [
    {
      "data_headers": {
        "roles": [
            {
                "source": {
                    "coord": {
                      "row": 4,
                      "col": 4
                    },
                    "direction": "right"
                },
                "role": "date"
            }
        ],
        "data_direction": "down",
        "data_cells": {
          "coord": {
            "row": 5,
            "col": 4
          },
          "direction": "right"
        }
      },
      "titles": {
        "source": {
          "coord": {
            "row": 5,
            "col": 0
          },
          "direction": "down"
        },
        "extract_params": {
          "default": "EMPTY"
        }
      },
      "metrics": {
        "source": {
          "coord": {
            "row": 5,
            "col": 3
          },
          "direction": "down"
        }
      },
      "title_ids": [],
      "dimensions": [
        {
          "name": "Publisher",
          "source": {
            "coord": {
              "row": 5,
              "col": 1
            },
            "direction": "down"
          }
        },
        {
          "name": "Platform",
          "source": {
            "coord": {
              "row": 5,
              "col": 2
            },
            "direction": "down"
          }
        }
      ],
      "kind": "non_counter.generic"
    }
  ]
}
