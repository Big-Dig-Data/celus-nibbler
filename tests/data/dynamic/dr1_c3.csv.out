"2021-01-01","2021-01-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-02-01","2021-02-28","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-03-01","2021-03-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","3"
"2021-04-01","2021-04-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-05-01","2021-05-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-06-01","2021-06-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-07-01","2021-07-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-08-01","2021-08-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-09-01","2021-09-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","1"
"2021-10-01","2021-10-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-11-01","2021-11-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","1"
"2021-12-01","2021-12-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-01-01","2021-01-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-02-01","2021-02-28","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-03-01","2021-03-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-04-01","2021-04-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-05-01","2021-05-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-06-01","2021-06-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-07-01","2021-07-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-08-01","2021-08-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-09-01","2021-09-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-10-01","2021-10-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-11-01","2021-11-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-12-01","2021-12-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-01-01","2021-01-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-02-01","2021-02-28","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","3"
"2021-03-01","2021-03-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","2"
"2021-04-01","2021-04-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-05-01","2021-05-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-06-01","2021-06-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-07-01","2021-07-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-08-01","2021-08-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-09-01","2021-09-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","2"
"2021-10-01","2021-10-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-11-01","2021-11-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-12-01","2021-12-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-01-01","2021-01-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-02-01","2021-02-28","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-03-01","2021-03-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-04-01","2021-04-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-05-01","2021-05-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-06-01","2021-06-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-07-01","2021-07-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-08-01","2021-08-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-09-01","2021-09-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-10-01","2021-10-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-11-01","2021-11-30","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-12-01","2021-12-31","","DB1","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-01-01","2021-01-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-02-01","2021-02-28","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-03-01","2021-03-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","2"
"2021-04-01","2021-04-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-05-01","2021-05-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-06-01","2021-06-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-07-01","2021-07-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-08-01","2021-08-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-09-01","2021-09-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","1"
"2021-10-01","2021-10-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-11-01","2021-11-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-12-01","2021-12-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total searches run","0"
"2021-01-01","2021-01-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-02-01","2021-02-28","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-03-01","2021-03-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-04-01","2021-04-30","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-05-01","2021-05-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-06-01","2021-06-30","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-07-01","2021-07-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-08-01","2021-08-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-09-01","2021-09-30","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-10-01","2021-10-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-11-01","2021-11-30","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-12-01","2021-12-31","","DB2","","","","","","Platform:Platform1|Publisher:","Searches-federated and automated","0"
"2021-01-01","2021-01-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","1"
"2021-02-01","2021-02-28","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","3"
"2021-03-01","2021-03-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","3"
"2021-04-01","2021-04-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-05-01","2021-05-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-06-01","2021-06-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-07-01","2021-07-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-08-01","2021-08-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-09-01","2021-09-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","1"
"2021-10-01","2021-10-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-11-01","2021-11-30","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","1"
"2021-12-01","2021-12-31","","DB2","","","","","","Platform:Platform1|Publisher:","Total sessions","0"
"2021-01-01","2021-01-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-02-01","2021-02-28","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-03-01","2021-03-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-04-01","2021-04-30","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-05-01","2021-05-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-06-01","2021-06-30","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-07-01","2021-07-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-08-01","2021-08-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-09-01","2021-09-30","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-10-01","2021-10-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-11-01","2021-11-30","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-12-01","2021-12-31","","DB2","","","","","","Platform:Platform1|Publisher:","Sessions-federated and automated","0"
"2021-01-01","2021-01-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-02-01","2021-02-28","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-03-01","2021-03-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-04-01","2021-04-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-05-01","2021-05-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-06-01","2021-06-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-07-01","2021-07-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-08-01","2021-08-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-09-01","2021-09-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-10-01","2021-10-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-11-01","2021-11-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-12-01","2021-12-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total searches run","0"
"2021-01-01","2021-01-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-02-01","2021-02-28","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-03-01","2021-03-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-04-01","2021-04-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-05-01","2021-05-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-06-01","2021-06-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-07-01","2021-07-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-08-01","2021-08-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-09-01","2021-09-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-10-01","2021-10-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-11-01","2021-11-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-12-01","2021-12-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Searches-federated and automated","0"
"2021-01-01","2021-01-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-02-01","2021-02-28","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","3"
"2021-03-01","2021-03-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","2"
"2021-04-01","2021-04-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-05-01","2021-05-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-06-01","2021-06-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-07-01","2021-07-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-08-01","2021-08-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-09-01","2021-09-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-10-01","2021-10-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","1"
"2021-11-01","2021-11-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","2"
"2021-12-01","2021-12-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Total sessions","0"
"2021-01-01","2021-01-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-02-01","2021-02-28","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-03-01","2021-03-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-04-01","2021-04-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-05-01","2021-05-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-06-01","2021-06-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-07-01","2021-07-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-08-01","2021-08-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-09-01","2021-09-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-10-01","2021-10-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-11-01","2021-11-30","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
"2021-12-01","2021-12-31","","DB3","","","","","","Platform:Platform1|Publisher:Publisher1","Sessions-federated and automated","0"
